doc/binning_basics.rst

```diff
@@ -0,0 +1,102 @@
+Using the binning module
+========================
+
+Stile contains some simple functions to bin your data.  Two of them--:class:`stile.BinStep
+<stile.binning.BinStep>` and :class:`stile.BinList <stile.binning.BinList>`--have simple,
+predefined ways of acting on your data set; the third, :class:`stile.BinFunction
+<stile.binning.BinFunction>`, uses user-defined functions to split your data.
+
+Basic interface
+---------------
+
+To start binning your data, you create a :class:`Bin*` object that will contain a binning
+definitions.  For instance, if you wanted to bin the ``ra`` column in 10 bins:
+
+>>> bin_object = stile.BinStep(field='ra', n_bins=10, low=0, high=360)
+
+To use it, call the object; it returns a list of objects which you can apply to your data to
+produce properly binned subsets.
+
+>>> for single_bin in bin_object():
+>>>     binned_data = single_bin(data)
+
+The ``single_bin`` above is actually another class called a :class:`stile.binning.SingleBin`.  It
+knows its boundaries and it also contains a string you can use in program outputs.
+
+>>> single_bin = bin_object()[0]
+>>> type(single_bin)
+>>> print single_bin.low
+0.0
+>>> print single_bin.hi
+36.0
+>>> print single_bin.short_name
+'0'
+
+Types of binning schemes
+------------------------
+
+The :class:`stile.BinList <stile.binning.BinList>` is the simplest class.  To create it, call it
+with a list of bin edges and a field name (see the :doc:`data` documentation for more information
+on field names).
+
+>>> bin_list_object = stile.BinList(field='g1', bin_list=[-10, -1, -0.5, -0.3, -0.1, -0.05, 0, 0.05, 0.1, 0.3, 0.5, 1, 10])
```

Comment:
Is there a way to get a line-break so it doesn't go off the side of the document in the PDF?